codes/classical/properties/cyclic/symmetric.yml

```diff
@@ -0,0 +1,22 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: symmetric
+physical: q-ary_digits
+logical: q-ary_digits
+
+name: 'Symmetric code'
+#introduced: ''
+
+description: 'A classical code \(C\) of length \(n\) over an alphabet is symmetric if, for each string \(c_1 c_2 \cdots c_n\in C\), the permuted string \(c_{\sigma(1)} c_{\sigma(2)} \cdots c_{\sigma(n)}\in C\).'
```

Comment:
This does not seem to make sense. This description does not specify which permutations sigma preserve the code; usually such a collection of permutations is called a permutation automorphism group of the code. Every code has some automorphism group, although for some codes it is trivial, i.e., contains only the identical permutation. 

I believe that what is required is a different class of codes, sometimes called Pless symmetry codes, see the book by Huffman and Pless, https://doi.org/10.1017/CBO9780511807077 Sec. 10.5.
If not, then the definition has to be made precise anyway.

Reply:
A symmetric code could be any code with a nontrivial automorphism group. But on 2nd thought this seems too broad. I propose we add the defn of automorphism group to the description of ecc.yml instead of making a new class. If you'd like, you can also try to find a notable class of codes with a particular group (note: pless is taken by another student). (BTW In quantum, automorphisms of the code would be logical gates, but I guess nobody uses those for gates in classical domain.)